src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -897,8 +890,19 @@ private void DoCloseHelper()
             //Raise Event
             RaiseRunspaceStateEvents();
 
-            // Report telemetry if we have no more open runspaces.
+            if (closeAllOpenRunspaces)
```

Comment:
This solution won't work well with ISE-like applications. Maybe we should open an issue to track it.
> I'm thinking about ISE here. Each tab is an interactive host with a different Runspace, so which one should be the default Runspace? 
A user can type `ctrl+t` and `ctrl+w` to create a new tab and close an existing tab, so it's very common that the first tab gets closed by `ctrl+w` later. If the first tab's Runspace is set to be the default, then closing that tab would terminate all tabs ...

Reply:
Yes, if there is no primary runspace then there is a problem with auto clean up.  In this case the static variable should not be used and the application needs to do its own cleanup on exit.  I don't think an issue needs to be created since this is just being used internally right now for PowerShell shell.

Reply:
>  I don't think an issue needs to be created since this is just being used internally right now for PowerShell shell.

Agreed. #Closed